pkg/yaml/yaml_test.go

```diff
@@ -0,0 +1,496 @@
+package yaml
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/kluctl/kluctl/v2/pkg/utils"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"testing/iotest"
+)
+
+type EmptyYamlConfig struct {
+}
+
+type SimpleYamlConfig struct {
+	Value string `yaml:"value"`
+}
+
+func TestReadYamlFile(t *testing.T) {
+	// Setup variables
+	existingEmptyYamlFileNamePattern := "test_read_yaml_file_existing_empty_*.yaml"
+	var existingEmptyYamlConf EmptyYamlConfig
+	nonExistingEmptyYamlFileName := "test_read_yaml_file_non_existing_empty.yaml"
+	var nonExistingEmptyYamlConf EmptyYamlConfig
+
+	// Create directory for test files
+	createTestDir()
+	defer cleanupTestDir()
+
+	// Setup temporary file
+	file, err := CreateTempFile(existingEmptyYamlFileNamePattern)
+	defer os.Remove(file.Name())
+	if err != nil {
+		t.Errorf("Can't create file: %s", file.Name())
+	}
+
+	//Read existing empty yaml file
+	existingEmptyYamlErr := ReadYamlFile(file.Name(), &existingEmptyYamlConf)
+	if existingEmptyYamlErr != nil {
+		t.Errorf("Can't read empty yaml file: %s", file.Name())
+	}
+	//Read non-existing empty yaml file
+	nonExistingEmptyYamlErr := ReadYamlFile(nonExistingEmptyYamlFileName, &nonExistingEmptyYamlConf)
+	if nonExistingEmptyYamlErr == nil {
+		t.Errorf("Should throw an error because %s doesn't exist", nonExistingEmptyYamlFileName)
+	}
+}
+
+func TestReadYamlAllFileFs(t *testing.T) {
```

Comment:
I assume that the "Fs" suffix is from an older incarnation of this test?
Also, can you add a test case with a non-empty yaml file that contains 2 docs?